Configuration for github actions

clbarnes opened the pull request:
See #227

ekohl commented:
I briefly pushed the branch to upstream which triggers the tests to actually run.

ekohl commented on .github/workflows/ci.yaml:
```diff
@@ -0,0 +1,31 @@
+on: [push, pull_request]
```
When it comes to lists, I'm always a fan of the multi line version
```suggestion
on:
  - push
  - pull_request
```

The same is even more important with `python-version` in the matrix , just so you can easily add or remove entries

ekohl commented on .github/workflows/ci.yaml:
```diff
@@ -0,0 +1,31 @@
+
+jobs:
+
+  tox:
```
I'd be nice if you could give this step a name.

clbarnes commented:
Thanks! MacOS fails because the image doesn't have mercurial in it by default. Other than that, seems to be passing on my side!

Before un-drafting this, I'll remove the appveyor and travis configs and add py39.

ekohl commented:
Just so you're aware of it: I now need to approve and run every time to actually run.

Could you also give it a name at the top level? That way we don't see the the filename but rather something nice. You can add `name: CI` at the very top.

clbarnes commented:
Don't worry about approving it so it can run here, I'm checking on my own branch and will un-draft when it's passing there. Working on the last failures!

clbarnes commented:
Green on my end. py39's argparse apparently formats differently to others', so I had to normalise the whitespace when doing string containment checks on those messages.

ekohl commented:
Perhaps you can squash the commits and leave out the Python 3.9 changes? Then I'd have no problem merging it but I don't have time to look at the 3.9 changes that closely.

ekohl reviewed (approved):
I can still see appveyor run. @c4urself can you check this too?

IMHO we should squash the commits (can do on merge, but we shouldn't forget).

clbarnes commented:
Updated the readme buttons, and changelog, and squashed. It's weird that appveyor is still doing anything; the repo doesn't have any mentions of it outside of the changelog. Maybe the repo is still registered on the appveyor side?

ekohl reviewed (approved):
That is also my theory but I don't know appveyor.

ekohl commented:
@florisla this has been open for a long time. Any reason we can't merge this?

florisla commented:
Much appreciated @clbarnes, this is great work!
